addons/test_event_full/tests/test_event_mail.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.event.tests.common import TestEventCommon
+from odoo.exceptions import ValidationError
+
+class TestTemplateRefModel(TestEventCommon):
+
+    def test_template_ref_model_constraint(self):
+
+        with self.assertRaises(ValidationError):
+            self.env['event.mail'].create({
```

Comment:
You should probably also test the inverse check (notification mail, sms template).